media-proxy/include/mesh/DEVELOPMENT.md

```diff
@@ -0,0 +1,195 @@
+# Development – New Architecture of Media Proxy to Support Multipoint Groups
+
+## Scope
+1. **`DONE`** Local connection (memif based)
+   * **`IN PROGRESS`** Local base class – `mesh::connection::Local`, derived from `mesh::connection::Connection`.
+      * **`IN PROGRESS`** Local Tx class – `mesh::connection::LocalTx`
+      * **`IN PROGRESS`** Local Rx class – `mesh::connection::LocalRx`
+1. RDMA connection
+   * RDMA base class – `mesh::connection::Rdma`, derived from `mesh::connection::Connection`.
+      * RDMA Tx class – `mesh::connection::RdmaTx`
+      * RDMA Rx class – `mesh::connection::RdmaRx`
+1. ST2110 connection
+   * ST2110 base class – `mesh::connection::ST2110`, derived from `mesh::connection::Connection`.
+      * ST2110 Tx base class – `mesh::connection::ST2110Tx`
+         * ST2110-20 Tx class – `mesh::connection::ST2110_20Tx`
+         * ST2110-22 Tx class – `mesh::connection::ST2110_22Tx`
+         * ST2110-30 Tx class – `mesh::connection::ST2110_30Tx`
+      * ST2110 Rx base class – `mesh::connection::ST2110Rx`
+         * ST2110-20 Rx class – `mesh::connection::ST2110_20Rx`
+         * ST2110-22 Rx class – `mesh::connection::ST2110_22Rx`
+         * ST2110-30 Rx class – `mesh::connection::ST2110_30Rx`
+
+## How It Is Supposed To Work?
+
+
+### Tx/Rx Test Scenario
+
+```mermaid
+flowchart LR
+    a[**Emulated Transmitter**
+      _derived from
+      Connection_]
+    b[**Emulated Receiver**
+      _derived from
+      Connection_]
+    tx[**ST2110_20Tx**
+       _derived from
+       Connection_]
+    rx[**ST2110_20Rx**
+       _derived from
+       Connection_]
+
+    subgraph Tx Side
+    a -- Link --> tx
+    end
+
+    tx -- Option A: Single node --> rx
+    tx -- Option B: Network / Physical cable --> rx
+
+    subgraph Rx Side
+    rx -- Link --> b
+    end
+```
+
+### Rx Side
+
+```cpp
+using namespace mesh;
+
+class EmulatedReceiver : public connection::Connection {
+public:
+    EmulatedReceiver() : _kind(connection::Kind::receiver) {}
+
+    connection::Result on_establish(context::Context& ctx) {
+        set_state(ctx, connection::State::active);
+    }
+
+    connection::Result on_shutdown(context::Context& ctx) {}
+
+    connection::Result on_receive(context::Context& ctx, void *ptr, uint32_t sz,
+                                  uint32_t& sent) {
+        printf("Received %s (%u)\n", (char *)ptr, sz);
+    }
+};
+
+int main()
+{
+    auto ctx = context::WithCancel(mesh::context::Background());
+    int res;
+
+    // Setup Emulated Receiver
+    auto emulated_rx = new EmulatedReceiver;
+    emulated_rx->establish(ctx);
+
+    // Setup Rx connection
+    auto conn_rx = new mesh::connection::ST2110_20Rx;
+
+    res = conn_rx->configure(ctx, some_ST2110_20Rx_related_config);
+    if (res) {
+        printf("Configure Rx failed: %s\n", mesh::connection::result2str(res));
+        goto exit;
```

Comment:
here you go from `Not Initialized` to shutdown, but your Lifecycle State Diagram does not show this flow.

I guess we need to be able to go from every state to `Deleting`

Reply:
Agree